### ISSUE TITLE
build script

### ISSUE THREAD

**Issue by kurumpa** [kurumpa/dotSwitcher#18]
to build the app without VS

**Comment by BladeMight** [comment]
What do you mean?
With other IDE's/Compilers?

**Comment by kurumpa** [comment]
With MSBuild tool that is a part of .NET framework, see https://github.com/kurumpa/dotSwitcher/blob/master/build_someday.cmd

**Comment by BladeMight** [comment]
Yes, it is possible just drag'n'drop your dotswitcher.sln to MSBuild.exe or you can create .bat file in the dotSwitcher project directory, copy-paste this:
```batch
@echo OFF
setlocal ENABLEEXTENSIONS
set KEY_NAME="HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\MSBuild\ToolsVersions"
set MSBUILD_PATH=""

for /f "tokens=3,* delims= " %%a in ('REG QUERY %KEY_NAME% /s /F MSBuildToolsPath') do (
  if exist "%%a\MSBuild.exe" (
    set MSBUILD_PATH="%%a\MSBuild.exe"
  )
)

if %MSBUILD_PATH%=="" (
  echo Msbuild path not found in registry
  pause
  exit /b 1
)
echo %MSBUILD_PATH%
%MSBUILD_PATH% dotSwitcher.sln
pause
```
RUN and it will compile it...

**Comment by kurumpa** [comment]
I still need to target any platform version ;) for now it is only .NET 4.5. Have to migrate also to .NET 2.0 or whatever version can be found in win7 by default

**Comment by BladeMight** [comment]
This depends on program code, i tried to dran'n'drop dotSwitcher to .NET 3.0 it sends me an error:
about LowLevelAdapter.cs & TrayIcon.cs are "Default parameter specifiers are not  permitted"
so you need first edit them, and change *Targeted .NET* in project Properties(In VS)

**Comment by kurumpa** [comment]
That's exactly what i was talking about, and it's not just changing the target, but also changing lots of code ;) i'll try to deal with selection converting first

**Comment by BladeMight** [comment]
I looked up on the source code and notice, that it requires **System.Linq** and **System.Threading.Tasks** that are avaible just in .NET 4.5, so if you will change code you need to find functions similar to the linked namespaces, from the one's avaible(e.g. .NET 2.0,.NET 3.5 etc)

**Comment by GremL1N** [comment]
Don't build with .Net 4.0 in my case.
Microsoft (R) Build Engine версии 4.6.1055.0
[Microsoft .NET Framework версии 4.0.30319.42000]
Error:
... not found "AxImp.exe" or .NET Framework SDK 1 not installed...
